tracing-subscriber/src/fmt/time/datetime.rs

```diff
@@ -4,16 +4,211 @@ use std::fmt;
+/// Copyright © 2003-2009 Bruce D. Evans or
+/// Copyright © 2008 Stephen L. Moshier or
+/// Copyright © 2017-2018 Arm Limited
+/// and labelled as such in comments in the individual source files. All
+/// have been licensed under extremely permissive terms.
+///
+/// The ARM memcpy code (src/string/arm/memcpy.S) is Copyright © 2008
+/// The Android Open Source Project and is licensed under a two-clause BSD
+/// license. It was taken from Bionic libc, used on Android.
+///
+/// The AArch64 memcpy and memset code (src/string/aarch64/*) are
+/// Copyright © 1999-2019, Arm Limited.
+///
+/// The implementation of DES for crypt (src/crypt/crypt_des.c) is
+/// Copyright © 1994 David Burren. It is licensed under a BSD license.
+///
+/// The implementation of blowfish crypt (src/crypt/crypt_blowfish.c) was
+/// originally written by Solar Designer and placed into the public
+/// domain. The code also comes with a fallback permissive license for use
+/// in jurisdictions that may not recognize the public domain.
+///
+/// The smoothsort implementation (src/stdlib/qsort.c) is Copyright © 2011
+/// Valentin Ochs and is licensed under an MIT-style license.
+///
+/// The x86_64 port was written by Nicholas J. Kain and is licensed under
+/// the standard MIT terms.
+///
+/// The mips and microblaze ports were originally written by Richard
+/// Pennington for use in the ellcc project. The original code was adapted
+/// by Rich Felker for build system and code conventions during upstream
+/// integration. It is licensed under the standard MIT terms.
+///
+/// The mips64 port was contributed by Imagination Technologies and is
+/// licensed under the standard MIT terms.
+///
+/// The powerpc port was also originally written by Richard Pennington,
+/// and later supplemented and integrated by John Spencer. It is licensed
+/// under the standard MIT terms.
+///
+/// All other files which have no copyright comments are original works
+/// produced specifically for use as part of this library, written either
+/// by Rich Felker, the main author of the library, or by one or more
+/// contibutors listed above. Details on authorship of individual files
+/// can be found in the git version control history of the project. The
+/// omission of copyright and license comments in each file is in the
+/// interest of source tree size.
+///
+/// In addition, permission is hereby granted for all public header files
+/// (include/* and arch/*/bits/*) and crt files intended to be linked into
+/// applications (crt/*, ldso/dlstart.c, and arch/*/crt_arch.h) to omit
+/// the copyright notice and permission notice otherwise required by the
+/// license, and to use these files without any requirement of
+/// attribution. These files include substantial contributions from:
+///
+/// Bobby Bingham
+/// John Spencer
+/// Nicholas J. Kain
+/// Rich Felker
+/// Richard Pennington
+/// Stefan Kristiansson
+/// Szabolcs Nagy
+///
+/// all of whom have explicitly granted such permission.
```

Comment:
nit, take it or leave it: typically these license notices are at the very top of the file, in non-documentation comments. in this case, though, this is a doc comment on a struct. perhaps it would be better to move the copyright notice? not a blocker either way.

Reply:
Thank you! Updated the PR as you suggested.